ui/src/criteria/attribute.tsx

```diff
@@ -81,17 +129,115 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+
+  const year = new Date().getFullYear();
+
+  if (isValid(attributeFilterHint?.integerBoundsHint)) {
+    const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
+
+    if (
+      integerBoundsHint?.max !== null &&
+      integerBoundsHint?.max !== undefined &&
+      integerBoundsHint?.min !== null &&
+      integerBoundsHint?.min !== undefined
+    ) {
+      const min = year - integerBoundsHint?.max;
+      const max = year - integerBoundsHint.min;
+
+      const [minValue, setMinValue] = React.useState<number>(min);
+      const [maxValue, setMaxValue] = React.useState<number>(max);
+
+      const updateValue = (newMin: number, newMax: number) => {
+        setMinValue(newMin);
+        setMaxValue(newMax);
+        props.dispatchFn(
+          produce(props.data, (data) => {
+            data.min = newMin;
+            data.max = newMax;
+          })
+        );
+      };
+
+      React.useEffect(() => updateValue(min, max), []);
```

Comment:
Updating the data store as part of rendering is generally a bad idea because it results in extra, unexpected updates. This can cause issues with debugging, performance, or features like undo/redo. There are two options:

1) Allow min/max to remain undefined and treat that as meaning unbounded. Once a change is made in the UI, that value would be stored and never return to the unbounded state. `generateFilter` would need to only add operands for defined limits.
2) Set correct initial values when the criteria is created.

In my opinion, option 2 is better because it's less surprising. Somebody working on the code in the future is much less likely to break it because they don't fully understand how it works.

With that in mind, you'll need to to:
1) Move the `CriteriaConfig` definition to `underlaysSlice.ts`. That makes sense anyways since that data would someday come from the backend.
2) Add the `Underlay` as a parameter to `createCriteria` in `cohort.ts` as well as the `initializeData` parameter of `registerCriteriaPlugin`. Update the cohort and attribute criteria to accept the new parameter.
3) In the attribute plugin, lookup the attribute in `initializeData` and set the min and max accordingly if it has an integer hint.